server.go

```diff
@@ -545,45 +518,36 @@ func (p *sshFxpFsetstatPacket) respond(svr *Server) responsePacket {
 		return statusFromError(p.ID, EBADF)
 	}
 
-	// additional unmarshalling is required for each possibility here
+	debug("fsetstat name \"%s\"", f.Name())
 	b := p.Attrs.([]byte)
-	var err error
+	return statusFromError(p.ID, applyAttrsToFile(f.Name(), p.Flags, b))
```

Comment:
Oh dear. :( I’m unsure of how this subtle change from operating on a handle to operating on the filename would work. We’ve run into issues before where performing operations on what should be an open handle but done to the filename, or on a filename but through an open/close block.

I know `Chtimes` only exists for the filename, and not on a handle… but we could at least attempt to isolate this operation.

Also know that various systems will open a file, then delete it. This is a really weird corner case, but it seems like everyone runs into this weird pattern eventually and get tripped up assuming that a file will always be accessible from its filename while it is currently open.